flink-core/src/main/java/org/apache/flink/util/TimeUtils.java

```diff
@@ -29,9 +33,23 @@
  */
 public class TimeUtils {
 
+	private static final Map<String, ChronoUnit> LABEL_TO_UNIT_MAP = Collections.unmodifiableMap(initMap());
+
 	/**
 	 * Parse the given string to a java {@link Duration}.
-	 * The string is like "123ms", "321s", "12min" and such.
+	 * The string is in format "{length value}{time unit label}", e.g. "123ms", "321 s".
+	 * If no time unit label is specified, it will be considered as milliseconds.
```

Comment:
Where is this default comes from?

How about fail with a message that time unit must be configured?

Reply:
TimeUtils previously support non-labeled duration parsed as millisecond duration.
I think we should keep it to not break anything which is already using it.

Reply:
Thanks for your explanation :-)